packages/assets-controllers/src/TokensController.ts

```diff
@@ -1013,6 +1038,29 @@ export class TokensController extends BaseController<
     return { newAllTokens, newAllIgnoredTokens, newAllDetectedTokens };
   }
 
+  #getAddressOrSelectedAddress(address: string | undefined): string {
+    if (address) {
+      return address;
+    }
+
+    // If the address is not defined (or empty), we fallback to the currently selected account's address
+    const selectedAccount = this.messagingSystem.call(
+      'AccountsController:getAccount',
+      this.#selectedAccountId,
+    );
+    return selectedAccount?.address || '';
```

Comment:
Let's create a new `#getSelectedAddress()` function for this (since we are using this pattern multiple times)